modules/homepage/src/main/frontend/src/themePage/Navbars/QuickSearchIdentifier.jsx

```diff
@@ -0,0 +1,97 @@
+import DescriptionIcon from "@material-ui/icons/Description";
+import AssignmentIndIcon from "@material-ui/icons/AssignmentInd";
+import AssignmentIcon from '@material-ui/icons/Assignment';
+import { Link } from "react-router-dom";
+import { getEntityIdentifier } from "../EntityIdentifier.jsx";
+
+// Location of the quick search result metadata in a node, outlining what needs to be highlighted
+const LFS_QUERY_MATCH_KEY = "lfs:queryMatch";
+const LFS_QUERY_MATCH_PATH_KEY = "@path";
+
+// Properties of the quick search result metadata
+const LFS_QUERY_QUESTION_KEY = "question";
+const LFS_QUERY_MATCH_BEFORE_KEY = "before";
+const LFS_QUERY_MATCH_TEXT_KEY = "text";
+const LFS_QUERY_MATCH_AFTER_KEY = "after";
+const LFS_QUERY_MATCH_NOTES_KEY = "inNotes";
+
+// Display how the query matched the result
+export function QuickSearchMatch(props) {
+    const { matchData, classes } = props;
+    // Adjust the question text to reflect the notes, if the match was on the notes
+    let questionText = matchData[LFS_QUERY_QUESTION_KEY] + (matchData[LFS_QUERY_MATCH_NOTES_KEY] ? " / Notes" : "");
+    return matchData && (
+      <React.Fragment>
+        <span className={classes.queryMatchKey}>{questionText}</span>
+        <span className={classes.queryMatchSeparator}>: </span>
+        <span className={classes.queryMatchBefore}>{matchData[LFS_QUERY_MATCH_BEFORE_KEY]}</span>
+        <span className={classes.highlightedText}>{matchData[LFS_QUERY_MATCH_TEXT_KEY]}</span>
+        <span className={classes.queryMatchAfter}>{matchData[LFS_QUERY_MATCH_AFTER_KEY]}</span>
+      </React.Fragment>
+    ) || null
+}
+
+function MatchAvatar(props) {
+    const { matchData, classes } = props;
+    let icon = <DescriptionIcon />;
+    let style = '';
+    switch (matchData["jcr:primaryType"]) {
+      case "lfs:Subject":
+        icon = <AssignmentIndIcon />;
+        style = { backgroundColor: green[500] };
+        break;
+      case "lfs:Questionnaire":
+        icon = <AssignmentIcon  />;
+        style = { backgroundColor: orange[500] };
+        break;
+      // default covers other cases
+      default:
+        icon = <DescriptionIcon />;
+        style = { backgroundColor: blue[500] };
+        break;
+    }
+    return <Avatar className={classes.searchResultAvatar} style={style}>{icon}</Avatar>;
+}
+
+  // Display a quick search result identifier with link to result section
+export function QuickSearchIdentifier(props) {
+    let { resultData, hideMatchInfo, classes } = props;
+    let anchorPath = resultData[LFS_QUERY_MATCH_KEY] ? resultData[LFS_QUERY_MATCH_KEY][LFS_QUERY_MATCH_PATH_KEY] : '';
+    let fullPath = `/content.html${resultData["@path"]}#${anchorPath}`;
+    if (resultData["jcr:primaryType"] == "lfs:Questionnaire") {
+      fullPath = `/content.html/admin${resultData["@path"]}#${anchorPath}`;
+    }
+    return (<ListItem className={classes.quickSearchResultsIdentifier}>
```

Comment:
We really don't need a class just to specify `alignItems`.
```suggestion
    return (<ListItem alignItems="flex-start">
```